.github/workflows/artifacts-build.yml

```diff
@@ -0,0 +1,75 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+##### README #####
+#
+# The CI action in this file is used to build the artifacts on pushes to a repository containing
+# the ICU4X service account key. All steps are skipped unless the key is present.
+#
+# If you are a frequent contributor, you can add the key to your fork. The key is shared with
+# icu4x collaborators and can be viewed here:
+#
+# https://drive.google.com/file/d/17-oMqRfuHOHL9hYp64NYOh8vcJ03DQHm/view
+#
+# To add the key, follow these steps:
+#
+# 1. Go to the secrets on your fork:
+#     - https://github.com/{USER}/icu4x/settings/secrets/actions
+# 2. Click "New repository secret" and enter the following information:
+#     - Name: ICU4X_GCP_SA_KEY
+#     - Value: The contents of the file linked above
+# 3. Click "Add secret"
+# 4. Re-run the latest "Artifacts Build" action on your fork to make sure it works:
+#     - https://github.com/{USER}/icu4x/actions/workflows/artifacts-build.yml
+
+name: Artifacts Build
+
+on:
+  push:
+    branches-ignore: ["main"]
+
+jobs:
+  credentials:
+    name: "Check Credentials"
+    runs-on: "ubuntu-latest"
+    env:
+      ICU4X_GCP_SA_KEY: "${{ secrets.ICU4X_GCP_SA_KEY }}"
+    steps:
+    - name: "Check for credentials"
+      run: |
+        if [ -z "$ICU4X_GCP_SA_KEY" ]
+        then
+          echo "GCP key not found. Docs previews will not be uploaded. If you are a frequent contributor, you may add the key to your fork; for instructions, see 'artifacts-build.yml'"
+          exit 1;
+        fi
+  docs:
+    name: "Docs Preview"
+    needs: credentials
+    runs-on: "ubuntu-latest"
+    env:
+      GCP_PROJECT_ID: "dev-infra-273822"
+      GCP_BUCKET_ID: "icu4x-pr-artifacts"
```

Comment:
the `project_id` is in the key and wouldn't need to be repeated here. Something like `export GCP_PROJECT_ID=$(echo "$ICU4X_GCP_SA_KEY" | jq -r .project_id)`.

Could also add `"icu4x_bucket_id": "icu4x-pr-artifacts"` to the key and retrieve it from there as well.

Reply:
The key is not *necessarily* JSON; it could be in some other format that doesn't have the project ID readily available. Also, I don't necessarily want to introduce the `jq` dependency.